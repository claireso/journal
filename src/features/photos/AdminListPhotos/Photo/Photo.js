import React, { useRef, useCallback } from 'react'
import PropTypes from 'prop-types'

import * as S from './Photo.styles'

import { PhotoTypes } from '@types'

import useInView from '@hooks/useInView'

import AnimatedImage from '@components/AnimatedImage'
import { IconPencil, IconDelete } from '@components/Icons'
import { ButtonIcon } from '@components/Buttons'

const Photo = ({ id, color, source, title, description, onEdit, onDelete }) => {
  const dom = useRef(null)
  const inView = useInView(dom)

  const handleEdit = useCallback(() => onEdit(id), [id, onEdit])
  const handleDelete = useCallback(() => onDelete(id), [id, onDelete])

  return (
    <S.PhotoWrapper ref={dom} data-testid="photo">
      <S.PhotoPicture style={{ color: color || 'transparent' }}>
        {inView && <AnimatedImage src={source} />}
      </S.PhotoPicture>
      <S.PhotoInner>
        {title && <S.PhotoTitle dangerouslySetInnerHTML={{ __html: title }} />}
        {description && <S.PhotoDescription>{description}</S.PhotoDescription>}
      </S.PhotoInner>
      <S.PhotoTools>
        <ButtonIcon data-testid="button-edit" onClick={handleEdit} title="Edit">
          <IconPencil />
        </ButtonIcon>
        <ButtonIcon data-testid="button-delete" onClick={handleDelete} title="Delete">
          <IconDelete />
        </ButtonIcon>
      </S.PhotoTools>
    </S.PhotoWrapper>
  )
}

Photo.propTypes = {
  ...PhotoTypes,
  onEdit: PropTypes.func.isRequired,
  onDelete: PropTypes.func.isRequired
}

export default Photo
