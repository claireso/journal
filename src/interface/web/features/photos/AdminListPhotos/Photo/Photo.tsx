import React from 'react'

import * as cls from './styles.css'

import { PhotoDto } from '@dto'
import { formatDate } from '@utils/date'

import Media from './Media'
import Actions from './Actions'

import Text from '@web/components/Text'

type PhotoProps = PhotoDto

const Photo = ({ id, color, media, title, description, created_at }: PhotoProps) => {
  return (
    <div className={cls.wrapper}>
      <Media {...media} color={color}>
        <Actions photoId={id} />
      </Media>
      <div className={cls.content}>
        <div className={cls.contentInner}>
          {title && (
            <Text as="span" size="sm">
              <span dangerouslySetInnerHTML={{ __html: title }} />
            </Text>
          )}
          {description && (
            <Text as="span" size="sm" color="neutral">
              <span dangerouslySetInnerHTML={{ __html: description }} />
            </Text>
          )}
        </div>
        <Text as="span" size="xxs" color="neutral" className={cls.date}>
          {formatDate(created_at)}
        </Text>
      </div>
    </div>
  )
}

export default Photo
