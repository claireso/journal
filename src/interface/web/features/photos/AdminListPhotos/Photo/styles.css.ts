import { pagesStyle, tokens, radius, py, px, radiusBottom, gap } from '@web/theme'

export const wrapper = pagesStyle({
  display: 'flex',
  flexDirection: 'column',
  minHeight: '100%',
  boxShadow: `3px 3px 3px ${tokens.colors['slate-200']}`,
  border: `1px solid ${tokens.colors.neutral.extralight}`,
  transition: 'border 150ms ease-out',
  ...radius('size-1'),
  ':hover': {
    borderColor: tokens.colors.primary.light
  }
})

export const content = pagesStyle({
  display: 'flex',
  flex: 1,
  justifyContent: 'space-between',
  flexDirection: 'column',
  background: tokens.colors.white,
  ...py('size-3'),
  ...px('size-3'),
  ...gap('size-0.5'),
  ...radiusBottom('size-1')
})

export const contentInner = pagesStyle({
  display: 'flex',
  flex: 1,
  flexDirection: 'column',
  ...gap('size-0.5')
})

export const date = pagesStyle({
  textAlign: 'right'
})

// export const actions = pagesStyle({
//   position: 'absolute',
//   top: 0,
//   left: 0,
//   right: 0,
//   bottom: 0,
//   // background: 'rgba(255, 255, 255, 0.7)',
//   background: `color-mix(in srgb, ${tokens.colors.neutral['2extralight']} 75%, transparent)`,
//   display: 'flex',
//   alignItems: 'center',
//   justifyContent: 'center',
//   visibility: 'hidden',
//   opacity: 0,
//   transition: 'opacity 250ms ease-out',
//   willChange: 'opacity',
//   ...gap('size-2'),
//   selectors: {
//     [`${wrapper}:hover &`]: {
//       visibility: 'visible',
//       opacity: 1
//     }
//   }
// })
