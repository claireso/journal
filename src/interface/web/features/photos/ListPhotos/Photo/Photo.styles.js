import { styled, theme } from '@web/theme'

import AnimatedImage from '@web/components/AnimatedImage'

const PORTRAIT = 'portrait'
const LANDSCAPE = 'landscape'
const SQUARE = 'square'
// const POSITION_LEFT = 'left'
const POSITION_CENTER = 'center'
const POSITION_RIGHT = 'right'

const configImages = {
  [PORTRAIT]: {
    cellWidth: {
      large: 4,
      small: 5
    }
  },
  [LANDSCAPE]: {
    cellWidth: {
      large: 8,
      small: 6
    }
  },
  [SQUARE]: {
    cellWidth: {
      large: 6,
      small: 6
    }
  }
}

const getColumnStart = (position = '', cellWidth) => {
  switch (position) {
    case 'center': {
      return (theme.sizes.gridColumns.value - cellWidth) / 2 + 1
    }
    case 'right': {
      return theme.sizes.gridColumns.value - cellWidth
    }
  }
}

export const Figure = styled('figure', {
  $$shadowWidth: '8rem',
  $$shadowWidthXs: '2rem',
  m: 0,
  gridColumnStart: 1,
  gridColumnEnd: `span ${configImages[LANDSCAPE].cellWidth.small}`,

  '@lg': {
    gridColumnEnd: `span ${configImages[LANDSCAPE].cellWidth.large}`,
    gridColumnStart: 2,
    '& + &': {
      mt: '20rem'
    }
  },

  variants: {
    position: {
      // landscape position
      [POSITION_CENTER]: {
        '@lg': {
          gridColumnStart: getColumnStart(POSITION_CENTER, configImages[LANDSCAPE].cellWidth.large)
        }
      },
      [POSITION_RIGHT]: {
        '@lg': {
          gridColumnStart: getColumnStart(POSITION_RIGHT, configImages[LANDSCAPE].cellWidth.large)
        }
      }
    },
    // width of portrait
    portrait: {
      true: {
        gridColumnEnd: `span ${configImages[PORTRAIT].cellWidth.small}`,
        '@lg': {
          gridColumnEnd: `span ${configImages[PORTRAIT].cellWidth.large}`
        }
      }
    },
    // width of square
    square: {
      true: {
        gridColumnEnd: `span ${configImages[SQUARE].cellWidth.small}`,
        '@lg': {
          gridColumnEnd: `span ${configImages[SQUARE].cellWidth.large}`
        }
      }
    },
    // box shadow to highlight pictures
    withColor: {
      true: {
        mt: '$$shadowWidthXs',
        '> div': {
          boxShadow: '0 0 0 $$shadowWidthXs currentColor',
          mb: 'calc(1rem + $$shadowWidthXs)',
          position: 'relative',
          '@sm': {
            '&::after': {
              background: 'currentColor',
              bottom: '-$5',
              content: '',
              display: 'block',
              left: '100%',
              position: 'absolute',
              top: '-$5',
              width: '99999px'
            }
          }
        },
        '@lg': {
          mt: '$$shadowWidth',
          '> div': {
            boxShadow: '0 0 0 $$shadowWidth currentColor',
            mb: 'calc(1rem + $$shadowWidth)'
          },
          'figure + &': {
            mt: 'calc(20rem + $$shadowWidth)'
          }
        }
      }
    }
  },

  // start position in grid
  compoundVariants: [
    // square position center
    {
      square: true,
      position: POSITION_CENTER,
      css: {
        '@lg': {
          gridColumnStart: getColumnStart(POSITION_CENTER, configImages[SQUARE].cellWidth.large)
        }
      }
    },
    // square position right
    {
      square: true,
      position: POSITION_RIGHT,
      css: {
        '@lg': {
          gridColumnStart: getColumnStart(POSITION_RIGHT, configImages[SQUARE].cellWidth.large)
        }
      }
    },
    // portrait position center
    {
      portrait: true,
      position: POSITION_CENTER,
      css: {
        '@lg': {
          gridColumnStart: getColumnStart(POSITION_CENTER, configImages[PORTRAIT].cellWidth.large)
        }
      }
    },
    // portrait position right
    {
      portrait: true,
      position: POSITION_RIGHT,
      css: {
        '@lg': {
          gridColumnStart: getColumnStart(POSITION_RIGHT, configImages[PORTRAIT].cellWidth.large)
        }
      }
    }
  ]
})

export const Title = styled('figcaption', {
  fontSize: '$3'
})

export const Description = styled('span', {
  color: '$gray400',
  display: 'block',
  fontSize: '$1',
  fontStyle: 'italic',
  mt: '$1'
})

export const PictureWrapper = styled('div', {
  background: 'currentColor',
  width: '100%',
  margin: '0 0 1rem',
  aspectRatio: 'var(--aspect-ratio)',

  '@supports (not (aspect-ratio: 1 / 1)) and (not (aspect-ratio: 3 / 2)) and (not (aspect-ratio: 2 / 3))': {
    position: 'relative',
    pt: 'calc(100% / var(--aspect-ratio))'
  }
})

export const Picture = styled(AnimatedImage, {
  display: 'block',
  width: '100%',
  m: 0,

  '@supports (not (aspect-ratio: 1 / 1)) and (not (aspect-ratio: 3 / 2)) and (not (aspect-ratio: 2 / 3))': {
    height: '100%',
    position: 'absolute',
    top: 0,
    left: 0
  }
})
