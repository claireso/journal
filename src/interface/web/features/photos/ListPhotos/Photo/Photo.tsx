'use client'

import React, { useRef } from 'react'
import type { PhotoDto } from '@dto'

import AnimatedImage from '@web/components/AnimatedImage'
import Text from '@web/components/Text'
import useInView from '@web/hooks/useInView'
import { tokens } from '@web/theme/core/tokens.css'

import * as cls from './styles.css'

interface PhotoProps extends PhotoDto {
  row: number
}

const Photo = ({ title, description, media, position, color, row }: PhotoProps) => {
  const dom = useRef<HTMLElement>(null!)
  const isInView = useInView(dom)

  return (
    <figure
      className={cls.figure({
        landscape: !media.portrait && !media.square,
        portrait: media.portrait,
        square: media.square,
        position: position,
        highlight: !!color
      })}
      ref={dom}
      style={{
        gridRowStart: row
      }}
    >
      <div
        className={cls.pictureWrapper({
          portrait: media.portrait,
          square: media.square,
          highlight: !!color
        })}
        style={{
          color: color || tokens.colors.neutral['2extralight']
        }}
      >
        {isInView && <AnimatedImage className={cls.picture} src={media.source} />}
      </div>
      <figcaption className={cls.figcaption}>
        {title && (
          <Text as="span" size="sm">
            <span dangerouslySetInnerHTML={{ __html: title }} />
          </Text>
        )}
        {description && (
          <Text as="span" size="xs" color="neutral" italic>
            <span dangerouslySetInnerHTML={{ __html: description }} />
          </Text>
        )}
      </figcaption>
    </figure>
  )
}

export default Photo
