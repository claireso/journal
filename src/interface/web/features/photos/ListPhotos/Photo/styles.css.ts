import { recipe } from '@vanilla-extract/recipes'
import { pagesStyle, responsiveStyle, tokens, gapY, mb } from '@web/theme'

const PORTRAIT = 'portrait'
const LANDSCAPE = 'landscape'
const SQUARE = 'square'
const POSITION_LEFT = 'left'
const POSITION_CENTER = 'center'
const POSITION_RIGHT = 'right'

const configImages = {
  [PORTRAIT]: {
    cellWidth: {
      large: 4,
      small: 5
    }
  },
  [LANDSCAPE]: {
    cellWidth: {
      large: 8,
      small: 6
    }
  },
  [SQUARE]: {
    cellWidth: {
      large: 6,
      small: 6
    }
  }
}

const getColumnStart = (position = '', cellWidth: number) => {
  const gridColumns = 12
  switch (position) {
    case 'center': {
      return (gridColumns - cellWidth) / 2 + 1
    }
    case 'right': {
      return gridColumns - cellWidth
    }
  }
}

export const figure = recipe({
  base: pagesStyle({
    ...responsiveStyle({
      md: {
        selectors: {
          '& + &': {
            marginTop: '20rem'
          }
        }
      }
    })
  }),
  variants: {
    landscape: {
      true: pagesStyle({
        gridColumnEnd: `span ${configImages[LANDSCAPE].cellWidth.small}`,
        ...responsiveStyle({
          md: {
            gridColumnEnd: `span ${configImages[LANDSCAPE].cellWidth.large}`
          }
        })
      })
    },
    portrait: {
      true: pagesStyle({
        gridColumnEnd: `span ${configImages[PORTRAIT].cellWidth.small}`,
        ...responsiveStyle({
          md: {
            gridColumnEnd: `span ${configImages[PORTRAIT].cellWidth.large}`
          }
        })
      })
    },
    square: {
      true: pagesStyle({
        gridColumnEnd: `span ${configImages[SQUARE].cellWidth.small}`,
        ...responsiveStyle({
          md: {
            gridColumnEnd: `span ${configImages[SQUARE].cellWidth.large}`
          }
        })
      })
    },
    position: {
      [POSITION_LEFT]: pagesStyle({
        gridColumnStart: 1,
        ...responsiveStyle({
          md: {
            gridColumnStart: 2
          }
        })
      }),
      [POSITION_CENTER]: {},
      [POSITION_RIGHT]: {}
    },
    highlight: {
      true: pagesStyle({
        marginTop: tokens.borders['border-width-photo'],
        ...responsiveStyle({
          md: {
            marginTop: tokens.borders['border-width-photo-md'],
            selectors: {
              '& + &': {
                marginTop: `calc(20rem + ${tokens.borders['border-width-photo-md']})`
              }
            }
          }
        })
      })
    }
  },
  compoundVariants: [
    {
      variants: {
        landscape: true,
        position: POSITION_CENTER
      },
      style: pagesStyle({
        ...responsiveStyle({
          md: {
            gridColumnStart: getColumnStart(POSITION_CENTER, configImages[LANDSCAPE].cellWidth.large)
          }
        })
      })
    },
    {
      variants: {
        landscape: true,
        position: POSITION_RIGHT
      },
      style: pagesStyle({
        ...responsiveStyle({
          md: {
            gridColumnStart: getColumnStart(POSITION_RIGHT, configImages[LANDSCAPE].cellWidth.large)
          }
        })
      })
    },
    {
      variants: {
        portrait: true,
        position: POSITION_CENTER
      },
      style: pagesStyle({
        ...responsiveStyle({
          md: {
            gridColumnStart: getColumnStart(POSITION_CENTER, configImages[PORTRAIT].cellWidth.large)
          }
        })
      })
    },
    {
      variants: {
        portrait: true,
        position: POSITION_RIGHT
      },
      style: pagesStyle({
        ...responsiveStyle({
          md: {
            gridColumnStart: getColumnStart(POSITION_RIGHT, configImages[PORTRAIT].cellWidth.large)
          }
        })
      })
    },
    {
      variants: {
        square: true,
        position: POSITION_CENTER
      },
      style: pagesStyle({
        ...responsiveStyle({
          md: {
            gridColumnStart: getColumnStart(POSITION_CENTER, configImages[SQUARE].cellWidth.large)
          }
        })
      })
    },
    {
      variants: {
        square: true,
        position: POSITION_RIGHT
      },
      style: pagesStyle({
        ...responsiveStyle({
          md: {
            gridColumnStart: getColumnStart(POSITION_RIGHT, configImages[SQUARE].cellWidth.large)
          }
        })
      })
    }
  ]
})

export const pictureWrapper = recipe({
  base: pagesStyle({
    background: 'currentcolor',
    ...mb('size-2')
  }),
  variants: {
    highlight: {
      true: pagesStyle({
        boxShadow: '0 0 0 2rem currentColor',
        marginBottom: `calc(1rem + ${tokens.borders['border-width-photo']})`,
        position: 'relative',
        selectors: {
          '&::after': {
            background: 'currentColor',
            bottom: `calc(${tokens.sizes['size-5']}*-1)`,
            content: '',
            display: 'block',
            left: '100%',
            position: 'absolute',
            top: `calc(${tokens.sizes['size-5']}*-1)`,
            width: '99999px'
          }
        },
        ...responsiveStyle({
          md: {
            boxShadow: '0 0 0 8rem currentColor',
            marginBottom: `calc(1rem + ${tokens.borders['border-width-photo-md']})`,
            selectors: {
              '&::after': {
                display: 'none'
              }
            }
          }
        })
      })
    }
  }
})

export const figcaption = pagesStyle({
  display: 'flex',
  flexDirection: 'column',
  ...gapY('size-1')
})
