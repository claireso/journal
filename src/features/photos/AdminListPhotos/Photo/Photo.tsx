import React, { useRef, useCallback } from 'react'

import * as S from './Photo.styles'

import { EnhancedPhoto } from '@models'
import useInView from '@hooks/useInView'

import AnimatedImage from '@components/AnimatedImage'
import { IconPencil, IconDelete } from '@components/Icons'
import { ButtonIcon } from '@components/Buttons'

interface PhotoProps extends EnhancedPhoto {
  onEdit: (id: number) => void
  onDelete: (id: number) => void
}

const Photo = ({ id, color, source, title, description, onEdit, onDelete }: PhotoProps) => {
  const dom = useRef(null)
  const inView = useInView(dom)

  const handleEdit = useCallback(() => onEdit(id), [id, onEdit])
  const handleDelete = useCallback(() => onDelete(id), [id, onDelete])

  return (
    <S.PhotoWrapper ref={dom} data-testid="photo">
      <S.PhotoPicture style={{ color: color || 'transparent' }}>
        {inView && <AnimatedImage src={source} />}
      </S.PhotoPicture>
      <S.PhotoInner>
        {title && <S.PhotoTitle dangerouslySetInnerHTML={{ __html: title }} />}
        {description && <S.PhotoDescription>{description}</S.PhotoDescription>}
      </S.PhotoInner>
      <S.PhotoTools>
        <ButtonIcon data-testid="button-edit" onClick={handleEdit} title="Edit">
          <IconPencil />
        </ButtonIcon>
        <ButtonIcon data-testid="button-delete" onClick={handleDelete} title="Delete">
          <IconDelete />
        </ButtonIcon>
      </S.PhotoTools>
    </S.PhotoWrapper>
  )
}

export default Photo
