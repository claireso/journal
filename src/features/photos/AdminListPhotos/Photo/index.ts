import Photo from './Photo'

export default Photo
