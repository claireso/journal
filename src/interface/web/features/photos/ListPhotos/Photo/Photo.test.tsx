// eslint-disable-next-line
// @ts-nocheck
import { render } from '@testing-library/react'

import Photo from './index'

describe('<Photo />', () => {
  const renderPhoto = (props = {}) => render(<Photo row={0} {...props} />)

  it('should render photo (landscape center)', () => {
    const { asFragment } = renderPhoto(global.__PHOTO__)

    expect(asFragment()).toMatchSnapshot()
  })

  it('should render photo (landscape left)', () => {
    const { asFragment } = renderPhoto({ ...global.__PHOTO__, position: 'left' })

    expect(asFragment()).toMatchSnapshot()
  })

  it('should render photo (landscape right)', () => {
    const { asFragment } = renderPhoto({ ...global.__PHOTO__, position: 'right' })

    expect(asFragment()).toMatchSnapshot()
  })

  it('should render photo (portrait center)', () => {
    const { asFragment } = renderPhoto({ ...global.__PHOTO__, media: { ...global.__PHOTO__.media, portrait: true } })

    expect(asFragment()).toMatchSnapshot()
  })

  it('should render photo (square center)', () => {
    const { asFragment } = renderPhoto({ ...global.__PHOTO__, media: { ...global.__PHOTO__.media, square: true } })

    expect(asFragment()).toMatchSnapshot()
  })
})
