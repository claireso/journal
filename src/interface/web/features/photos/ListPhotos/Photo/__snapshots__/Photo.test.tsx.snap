// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Photo /> should render photo (landscape center) 1`] = `
<DocumentFragment>
  <figure
    class="styles__3b5bfuc styles_layerStyle__3b5bfu0 styles_figure_landscape_true__3b5bfud styles_layerStyle__3b5bfu1 styles_figure_position_center__3b5bfuh styles_layerStyle__3b5bfu6"
    style="grid-row-start: 0;"
  >
    <div
      class="styles__3b5bfuo styles_layerStyle__3b5bfuk"
    />
    <figcaption
      class="styles_layerStyle__3b5bfut"
    >
      <span
        class="styles_text__kgzpwem styles_text_size_sm__kgzpwep styles_layerStyle__kgzpwe2 styles_text_color_base__kgzpwev styles_layerStyle__kgzpwe8 styles_text_weight_normal__kgzpwe13 styles_layerStyle__kgzpweg"
      >
        <span>
          Single photography
        </span>
      </span>
      <span
        class="styles_text__kgzpwem styles_text_size_xs__kgzpweo styles_layerStyle__kgzpwe1 styles_text_color_neutral__kgzpwex styles_layerStyle__kgzpwea styles_text_weight_normal__kgzpwe13 styles_layerStyle__kgzpweg styles_text_italic_true__kgzpwez styles_layerStyle__kgzpwec"
      >
        <span>
          Janvier 2019
        </span>
      </span>
    </figcaption>
  </figure>
</DocumentFragment>
`;

exports[`<Photo /> should render photo (landscape left) 1`] = `
<DocumentFragment>
  <figure
    class="styles__3b5bfuc styles_layerStyle__3b5bfu0 styles_figure_landscape_true__3b5bfud styles_layerStyle__3b5bfu1 styles_figure_position_left__3b5bfug styles_layerStyle__3b5bfu4"
    style="grid-row-start: 0;"
  >
    <div
      class="styles__3b5bfuo styles_layerStyle__3b5bfuk"
    />
    <figcaption
      class="styles_layerStyle__3b5bfut"
    >
      <span
        class="styles_text__kgzpwem styles_text_size_sm__kgzpwep styles_layerStyle__kgzpwe2 styles_text_color_base__kgzpwev styles_layerStyle__kgzpwe8 styles_text_weight_normal__kgzpwe13 styles_layerStyle__kgzpweg"
      >
        <span>
          Single photography
        </span>
      </span>
      <span
        class="styles_text__kgzpwem styles_text_size_xs__kgzpweo styles_layerStyle__kgzpwe1 styles_text_color_neutral__kgzpwex styles_layerStyle__kgzpwea styles_text_weight_normal__kgzpwe13 styles_layerStyle__kgzpweg styles_text_italic_true__kgzpwez styles_layerStyle__kgzpwec"
      >
        <span>
          Janvier 2019
        </span>
      </span>
    </figcaption>
  </figure>
</DocumentFragment>
`;

exports[`<Photo /> should render photo (landscape right) 1`] = `
<DocumentFragment>
  <figure
    class="styles__3b5bfuc styles_layerStyle__3b5bfu0 styles_figure_landscape_true__3b5bfud styles_layerStyle__3b5bfu1 styles_figure_position_right__3b5bfui styles_layerStyle__3b5bfu7"
    style="grid-row-start: 0;"
  >
    <div
      class="styles__3b5bfuo styles_layerStyle__3b5bfuk"
    />
    <figcaption
      class="styles_layerStyle__3b5bfut"
    >
      <span
        class="styles_text__kgzpwem styles_text_size_sm__kgzpwep styles_layerStyle__kgzpwe2 styles_text_color_base__kgzpwev styles_layerStyle__kgzpwe8 styles_text_weight_normal__kgzpwe13 styles_layerStyle__kgzpweg"
      >
        <span>
          Single photography
        </span>
      </span>
      <span
        class="styles_text__kgzpwem styles_text_size_xs__kgzpweo styles_layerStyle__kgzpwe1 styles_text_color_neutral__kgzpwex styles_layerStyle__kgzpwea styles_text_weight_normal__kgzpwe13 styles_layerStyle__kgzpweg styles_text_italic_true__kgzpwez styles_layerStyle__kgzpwec"
      >
        <span>
          Janvier 2019
        </span>
      </span>
    </figcaption>
  </figure>
</DocumentFragment>
`;

exports[`<Photo /> should render photo (portrait center) 1`] = `
<DocumentFragment>
  <figure
    class="styles__3b5bfuc styles_layerStyle__3b5bfu0 styles_figure_landscape_true__3b5bfud styles_layerStyle__3b5bfu1 styles_figure_position_center__3b5bfuh styles_layerStyle__3b5bfu6"
    style="grid-row-start: 0;"
  >
    <div
      class="styles__3b5bfuo styles_layerStyle__3b5bfuk"
    />
    <figcaption
      class="styles_layerStyle__3b5bfut"
    >
      <span
        class="styles_text__kgzpwem styles_text_size_sm__kgzpwep styles_layerStyle__kgzpwe2 styles_text_color_base__kgzpwev styles_layerStyle__kgzpwe8 styles_text_weight_normal__kgzpwe13 styles_layerStyle__kgzpweg"
      >
        <span>
          Single photography
        </span>
      </span>
      <span
        class="styles_text__kgzpwem styles_text_size_xs__kgzpweo styles_layerStyle__kgzpwe1 styles_text_color_neutral__kgzpwex styles_layerStyle__kgzpwea styles_text_weight_normal__kgzpwe13 styles_layerStyle__kgzpweg styles_text_italic_true__kgzpwez styles_layerStyle__kgzpwec"
      >
        <span>
          Janvier 2019
        </span>
      </span>
    </figcaption>
  </figure>
</DocumentFragment>
`;

exports[`<Photo /> should render photo (square center) 1`] = `
<DocumentFragment>
  <figure
    class="styles__3b5bfuc styles_layerStyle__3b5bfu0 styles_figure_landscape_true__3b5bfud styles_layerStyle__3b5bfu1 styles_figure_position_center__3b5bfuh styles_layerStyle__3b5bfu6"
    style="grid-row-start: 0;"
  >
    <div
      class="styles__3b5bfuo styles_layerStyle__3b5bfuk"
    />
    <figcaption
      class="styles_layerStyle__3b5bfut"
    >
      <span
        class="styles_text__kgzpwem styles_text_size_sm__kgzpwep styles_layerStyle__kgzpwe2 styles_text_color_base__kgzpwev styles_layerStyle__kgzpwe8 styles_text_weight_normal__kgzpwe13 styles_layerStyle__kgzpweg"
      >
        <span>
          Single photography
        </span>
      </span>
      <span
        class="styles_text__kgzpwem styles_text_size_xs__kgzpweo styles_layerStyle__kgzpwe1 styles_text_color_neutral__kgzpwex styles_layerStyle__kgzpwea styles_text_weight_normal__kgzpwe13 styles_layerStyle__kgzpweg styles_text_italic_true__kgzpwez styles_layerStyle__kgzpwec"
      >
        <span>
          Janvier 2019
        </span>
      </span>
    </figcaption>
  </figure>
</DocumentFragment>
`;
